File: src/com/geo/GeoQuake/adapters/QuakeAdapter.java
package com.geo.GeoQuake.adapters;

import android.content.Context;
import android.graphics.Color;
import android.support.v4.content.ContextCompat;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.geo.GeoQuake.R;
import com.geo.GeoQuake.models.Earthquake;

import java.text.DateFormat;
import java.util.ArrayList;

import butterknife.Bind;
import butterknife.ButterKnife;

/**
 * Created by George Stinson on 2017-03-15.
 */
public class QuakeAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private ArrayList<Earthquake> mEarthquakes;
    OnQuakeItemClickedListener listener;

    public QuakeAdapter(OnQuakeItemClickedListener listener) {
        this.listener = listener;
    }

    public void setQuakeList(ArrayList<Earthquake> earthquakes) {
        this.mEarthquakes = earthquakes;
        notifyDataSetChanged();
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.quake_list_item_layout, parent, false);
        return new QuakeViewHolder(view, listener);
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
        ((QuakeViewHolder)holder).bind(mEarthquakes.get(position));
    }

    @Override
    public int getItemCount() {
        return mEarthquakes == null ? 0 : mEarthquakes.size();
    }

    public interface OnQuakeItemClickedListener {
        void onQuakeClicked(Earthquake earthquake);
        void onQuakeLongClick(Earthquake earthquake);
    }

    public class QuakeViewHolder extends RecyclerView.ViewHolder {
        @Bind(R.id.list_magnitude_text)
        TextView magnitudeTextView;
        @Bind(R.id.list_location_text)
        TextView locationTextView;
        @Bind(R.id.list_time_text)
        TextView timeTextViewHolder;
        OnQuakeItemClickedListener listener;

        public QuakeViewHolder(View itemView, OnQuakeItemClickedListener listener) {
            super(itemView);
            ButterKnife.bind(this, itemView);
            this.listener = listener;
        }

        public void bind(final Earthquake earthquake) {
            String magnitude = "";
            //deal with weird, negative magnitudes
            if(magnitude.equals("")) {
                if (earthquake.getMag() < 0.0) {
                    magnitude += 0.0;
                } else {
                    magnitude += earthquake.getMag(); //String var to get AS to quit complaining about concatenation
                }
                Context context = itemView.getContext();
                magnitudeTextView.setText(magnitude);
                magnitudeTextView.setTextColor(Color.WHITE);
                locationTextView.setText(earthquake.getPlace());
                if (earthquake.getMag() <= 1.00) {
                    magnitudeTextView.setTextColor(ContextCompat.getColor(context, R.color.material_green));
                } else if (earthquake.getMag() <= 2.50) {
                    magnitudeTextView.setTextColor(ContextCompat.getColor(context, R.color.material_orange));
                } else if (earthquake.getMag() <= 4.50) {
                    magnitudeTextView.setTextColor(ContextCompat.getColor(context, R.color.material_deeporange));
                } else {
                    magnitudeTextView.setTextColor(ContextCompat.getColor(context, R.color.material_red));
                }
                if(earthquake.getTime() != 0) {
                    java.text.DateFormat df = java.text.DateFormat.getDateTimeInstance(DateFormat.LONG, DateFormat.MEDIUM);
                    timeTextViewHolder.setText(df.format(earthquake.getTime()));
                } else {
                    timeTextViewHolder.setText(earthquake.getTimeString());
                }
                itemView.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        listener.onQuakeClicked(earthquake);
                    }
                });
                itemView.setOnLongClickListener(new View.OnLongClickListener() {
                    @Override
                    public boolean onLongClick(View v) {
                        listener.onQuakeLongClick(earthquake);
                        return true;
                    }
                });
            }
        }
    }
}

File: src/com/geo/GeoQuake/adapters/QuakeAdapter.kt
package com.geo.GeoQuake.adapters

import android.graphics.Color
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.geo.GeoQuake.R
import com.geo.GeoQuake.models.Earthquake
import java.text.DateFormat
import java.util.*

/**
 * Created by George Stinson on 2017-03-15.
 */
class QuakeAdapter(internal var listener: OnQuakeItemClickedListener) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    private var mEarthquakes: ArrayList<Earthquake>? = null

    fun setQuakeList(earthquakes: ArrayList<Earthquake>) {
        this.mEarthquakes = earthquakes
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.quake_list_item_layout, parent, false)
        return QuakeViewHolder(view, listener)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        (holder as QuakeViewHolder).bind(mEarthquakes!![position])
    }

    override fun getItemCount(): Int {
        return if (mEarthquakes == null) 0 else mEarthquakes!!.size
    }

    interface OnQuakeItemClickedListener {
        fun onQuakeClicked(earthquake: Earthquake)
        fun onQuakeLongClick(earthquake: Earthquake)
    }

    inner class QuakeViewHolder(itemView: View, internal var listener: OnQuakeItemClickedListener) : RecyclerView.ViewHolder(itemView) {

        internal var magnitudeTextView: TextView
        internal var locationTextView: TextView
        internal var timeTextViewHolder: TextView

        init {
            this.magnitudeTextView = itemView.findViewById(R.id.list_magnitude_text)
            this.locationTextView = itemView.findViewById(R.id.list_location_text)
            this.timeTextViewHolder = itemView.findViewById(R.id.list_time_text)
        }

        fun bind(earthquake: Earthquake) {
            var magnitude = ""
            //deal with weird, negative magnitudes
            if (magnitude == "") {
                magnitude += if (earthquake.mag < 0.0) {
                    0.0
                } else {
                    earthquake.mag
                }

                magnitudeTextView.text = String.format("%.2f", magnitude.toDouble())
                magnitudeTextView.setTextColor(Color.WHITE)
                locationTextView.text = earthquake.place

                setMagnitudeColor(earthquake.mag)

                if (earthquake.time != 0L) {
                    val df = DateFormat.getDateTimeInstance(DateFormat.LONG, DateFormat.MEDIUM)
                    timeTextViewHolder.text = df.format(earthquake.time)
                } else {
                    timeTextViewHolder.text = earthquake.timeString
                }
                itemView.setOnClickListener { listener.onQuakeClicked(earthquake) }
                itemView.setOnLongClickListener {
                    listener.onQuakeLongClick(earthquake)
                    true
                }
            }
        }

        /**
         * Set text color based on magnitude
         */
        private fun setMagnitudeColor(mag: Double) {
            when {
                mag <= 1.00 -> {
                    magnitudeTextView.setTextColor(ContextCompat.getColor(itemView.context, R.color.material_green))
                }
                mag <= 2.50 -> {
                    magnitudeTextView.setTextColor(ContextCompat.getColor(itemView.context, R.color.material_orange))
                }
                mag <= 4.50 -> {
                    magnitudeTextView.setTextColor(ContextCompat.getColor(itemView.context, R.color.material_deeporange))
                }
                else -> {
                    magnitudeTextView.setTextColor(ContextCompat.getColor(itemView.context, R.color.material_red))
                }
            }
        }
    }
}
